tree/ntuple/v7/inc/ROOT/RNTupleJoinTable.hxx

```diff
@@ -70,10 +70,11 @@ private:
    /// fields) to their respective entry numbers.
    std::unordered_map<RCombinedJoinFieldValue, std::vector<ROOT::NTupleSize_t>, RCombinedJoinFieldValueHash> fJoinTable;
 
-   /// The page source belonging to the RNTuple for which to build the join table.
-   std::unique_ptr<RPageSource> fPageSource;
+   /// Names of the join fields used for the mapping to their respective entry numbers.
+   std::vector<std::string> fJoinFieldNames;
 
-   /// The fields for which the join table is built. Used to compute the hashes for each entry value.
+   /// The fields for which the join table is built, corresponding to `fJoinFieldNames`. Used to compute the hashes for
+   /// each entry value.
    std::vector<std::unique_ptr<RFieldBase>> fJoinFields;
```

Comment:
Do we still need the fields as a member variable or can it become local to `Build()`?

Reply:
In `GetAllEntryNumbers`, we need to know the size of the field to be able to properly cast the incoming value pointers. This currently uses `RFieldBase::GetValueSize`. We can of course also only store the sizes themselves.